src/lib/y2network/connection/connection.rb

```diff
@@ -0,0 +1,20 @@
+require "y2network/interface"
+
+module Y2Network
+  module Connection
+    class Connection
+      # @return [Interface]
+      attr_accessor :interface
+      # @return [String] Bootproto (static, dhcp, none)
+      attr_accessor :bootproto
+      # @return [IPAddr]
+      attr_accessor :ip_address
+      # @return [Array<IPAddr>]
+      attr_accessor :secondary_ip_addresses
+      # @return [Integer]
+      attr_accessor :mtu
```

Comment:
same here, does mtu makes sense for all connections?

Reply:
While the `bootproto` is or seems to be clear a connection attribute, I was not so sure about the MTU. Probably it should be moved to the specialized classes that permits to establish the MTU

Reply:
MTU is related to physical layer (eth, ib, tokenring ;-) and do not make sense e.g. for vlans

Reply:
In nm-connection-editor.. the MTU is allowed per VLAN... BTW, currently we offer it in all the interfaces configuration, isn't it?

Reply:
yes, in general tab ... but it doesn't mean that it is correct.

vlan is defined at physical layer (you can see it as extension of physical layer) - especially tagged vlan's which extends (e.g.) ethernet frame with vlan id.

Reply:
It basically let the responsibility into the user